repository.go

```diff
@@ -64,35 +64,54 @@ func NewRepository() (*Repository, error) {
 		return nil, err
 	}
 
-	if _, err := client.Update(); err != nil {
+	r := &Repository{client: client}
+	if err := r.update(); err != nil {
 		return nil, err
 	}
-
-	r := &Repository{client: client}
 	go r.updateLoop()
 	return r, nil
 }
 
+func (r *Repository) Channels() []*Channel {
+	return r.channels.Load().([]*Channel)
+}
+
+var updateAttempts = attempt.Strategy{
+	Total: time.Minute,
+	Delay: 5 * time.Second,
+}
+
+func (r *Repository) updateLoop() {
+	interval := 5 * time.Minute
+
+	for range time.Tick(interval) {
+		if err := updateAttempts.Run(r.update); err != nil {
+			shutdown.Fatal(err)
```

Comment:
I'm not sure exiting is the right thing to do. It should just be a transient failure, serving stale data is fine. We could throw a last updated at timestamp in the JSON.